### PR TITLE
Refactor gen_defaults to support multiple generated code blocks in the same file independently.

### PR DESCRIPTION
**Pull request by darrenaustin**
The current `gen_defaults` script will only allow a single block of code to be appended to the end of a file. This PR refactors it such that each block is given a name and can be anywhere in the file and updated properly. This will allow us to have multiple generated blocks in a given file and have them updated independently. This in turn will make it easier to support multiple variants of tokens in a single file (say for different button variants).

In addition I added some more internal documentation as well as a new test to check the updating of multiple blocks in a file.

As part of this PR I regenerated all the existing code blocks to match the new format. While doing so, I took the liberty to  clean up the naming for the generated (FooDefaultsM3) and hand coded (FooDefaultsM2) FooThemeData classes so that they are consistent across all the widgets.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by guidezpl — APPROVED** [review]
Nice!

File: dev/tools/gen_defaults/test/gen_defaults_test.dart [hunk 1]
```diff
@@ -59,38 +62,166 @@ static final String tokenBar = 'Barfoo';
...
+
+// END GENERATED TOKEN PROPERTIES - Test
+''');
+
+    } finally {
+      tempDir.deleteSync(recursive: true);
+    }
+  });
+
+  test('Multiple templates can modify different code chunks in the same file', () {
```
Comment (HansMuller) [inline dev/tools/gen_defaults/test/gen_defaults_test.dart:110]: Why are we referring to blocks as "chunks" here?

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by QuncCccccc — APPROVED** [review]
LGTM!

**Pull request merged**